src/Toaster/TimerToast.tsx

```diff
@@ -12,21 +12,38 @@ export const Props = {
   style: t.maybe(t.Object)
 };
 
+export type TimerToastProps = {
+  children: JSX.Element,
+  onTimeout: (uniqueKey?: string) => void,
+  duration: number,
+  uniqueKey?: string,
+  id?: string,
+  className?: string,
+  style?: React.CSSProperties
+};
+
+export type TimerToastState = {
+  completed: boolean
+};
+
 @props(Props)
-export default class TimerToast extends React.Component {
+export default class TimerToast extends React.Component<TimerToastProps, TimerToastState> {
 
-  constructor(props) {
-    super(props);
-    this.state = { completed: false };
-  }
+  private timer: number | null;
+
+  state = { completed: false };
 
-  componentDidMount = () => this.resetTimer();
+  componentDidMount() {
+    this.resetTimer();
+  }
 
-  componentWillUnmount = () => this.clearTimer();
+  componentWillUnmount() {
+    this.clearTimer();
+  }
 
   resetTimer = () => {
     if (!this.timer && !this.state.completed) {
-      this.timer = setTimeout(this.onTimeout, this.props.duration);
+      this.timer = window.setTimeout(this.onTimeout, this.props.duration);
```

Comment:
this `window` is just for helping VSCode picking up the right overload (otherwise it thinks it's node's `setTimeout` which returns a `Timer` instead of a `number`.